graphql/src/graphql_schema.ml

```diff
@@ -1068,26 +1199,54 @@ end
     obj.name = type_condition ||
       List.exists (fun (abstract : abstract) -> abstract.name = type_condition) !(obj.abstracts)
 
-  let rec collect_fields : fragment_map -> ('ctx, 'src) obj -> Graphql_parser.selection list -> Graphql_parser.field list = fun fragment_map obj fields ->
+  let rec should_include ctx (directives : Graphql_parser.directive list) =
+    let open Rresult in
+    match directives with
+    | [] -> Ok true
+    | { name = "skip"; arguments }::rest ->
+        should_include ctx rest >>= fun should_include_rest ->
+          let Directive directive = skip_directive in
+          let resolver = directive.resolve in
+          Arg.eval_arglist ctx.variables directive.args arguments resolver
+          |> Rresult.R.map (fun skip -> not skip && should_include_rest)
+    | { name = "include"; arguments }::rest ->
+        should_include ctx rest >>= fun should_include_rest ->
+          let Directive directive = include_directive in
+          let resolver = directive.resolve in
+          Arg.eval_arglist ctx.variables directive.args arguments resolver
+          |> Rresult.R.map (fun includ -> includ && should_include_rest)
+    | { name; _ }::_ ->
+        let err = Format.sprintf "Unknown directive: %s" name in
+        Error err
+
+  let rec collect_fields : 'ctx execution_context -> ('ctx, 'src) obj -> Graphql_parser.selection list -> (Graphql_parser.field list, string) result =
+    fun ctx obj fields ->
+    let open Rresult in
     List.map (function
     | Graphql_parser.Field field ->
-        [field]
+        should_include ctx field.directives >>| fun should_include_field ->
```

Comment:
I would suggest giving the variable the short name and the function the longer name. Maybe the function `should_include_field` and the variable `include_field`?